dashboard/app/jobs/evaluate_rubric_job.rb

```diff
@@ -157,25 +218,36 @@ def self.get_lesson_s3_name(script_level)
     end
   end
 
-  private def write_ai_evaluations(user, ai_evaluations, rubric, channel_id, project_version)
-    _owner_id, project_id = storage_decrypt_channel_id(channel_id)
-
-    # record the ai evaluations to the database
-    # TODO: pass along and update the 'requester' to the correct id
+  private def write_ai_evaluations(user, ai_evaluations, rubric, rubric_ai_evaluation, project_version)
+    # Record the AI evaluations atomically to the database
     ActiveRecord::Base.transaction do
-      ai_evaluations.each do |evaluation|
-        learning_goal = rubric.learning_goals.all.find {|lg| lg.learning_goal == evaluation['Key Concept']}
-        understanding = understanding_s_to_i(evaluation['Grade'])
+      # Update the base rubric status
+      rubric_ai_evaluation.status = SharedConstants::RUBRIC_AI_EVALUATION_STATUS[:SUCCESS]
+
+      # Record the version of the code we evaluated
+      rubric_ai_evaluation.project_version = project_version
+
+      # Write out all of the learning goal records
+      ai_mapping = {}
+      ai_evaluations.each do |ai_evaluation|
+        ai_mapping[ai_evaluation['Key Concept']] = ai_evaluation
+      end
+
+      # For every learning goal in the rubric, find out if the AI has
+      # assessed it. If so, create a learning goal AI evaluation record.
```

Comment:
idea for potential follow-up: do something responsible if we did not receive ai evaluation for all the learning goals that are ai_enabled. raise an error? provide the teacher with the chance to rerun ai? explain in the UI why this particular result is missing? etc. I will capture this to follow up on, but interested in your thoughts